barretenberg/cpp/src/barretenberg/vm2/simulation/execution.test.cpp

```diff
@@ -69,7 +72,10 @@ TEST_F(ExecutionSimulationTest, Call)
 
     EXPECT_CALL(context, get_memory);
     EXPECT_CALL(context, get_address).WillRepeatedly(ReturnRef(parent_address));
-    EXPECT_CALL(memory, get).WillOnce(ReturnRef(nested_address_value));
+    EXPECT_CALL(memory, get)
+        .WillOnce(ReturnRef(l2_gas_allocated))      // l2_gas_offset
+        .WillOnce(ReturnRef(da_gas_allocated))      // da_gas_offset
+        .WillOnce(ReturnRef(nested_address_value)); // contract_address
```

Comment:
Hmm I don't remember exactly but I think you can do sth like
```
EXPECT_CALL(memory, get(addr1)).WillOnce(...)
EXPECT_CALL(memory, get(addr2)).WillOnce(...)
...
```

That would make this order-independent and more accurate if it is possible.

https://stackoverflow.com/questions/34885876/how-to-inspect-argument-to-a-gmock-expect-call